app/controllers/api/members_controller.rb

```diff
@@ -16,4 +22,13 @@ def create
   def member_params
     params.permit(:name, :email, :country, :postal)
   end
+
+  def member_validation
+    [
+      { name: 'email', data_type: 'email', required: true },
+      { name: 'country', data_type: 'country', required: false },
+      { name: 'postal', data_type: 'postal', required: false },
+      { name: 'name', data_type: 'text', required: true }
+    ]
+  end
```

Comment:
I have to respectfully disagree, controller actions should be super thin. Their only responsibility should be to get incoming params and respond, anything else should be delegated.